spec/unit/driver/auto_scaling_group_spec.rb

```diff
@@ -131,4 +135,47 @@
       expect(@driver.instance_statuses).to eq( { 'i-abcd1234' => returned_status } )
     end
   end
+
+  describe '#wait_for_desired_capacity' do
+    it 'completes if desired capacity reached' do
+      expect(@driver).to receive(:desired_capacity_reached?).and_return(true)
+
+      @driver.wait_for_desired_capacity
+    end
+
+    it 'times out if desired capacity is not reached' do
+      expect(@driver).to receive(:desired_capacity_reached?).and_return(false)
+
+      expect { @driver.wait_for_desired_capacity }.to raise_error(Timeout::Error)
+    end
+  end
+
+  describe '#desired_capacity_reached?' do
+    it 'returns true if healthy instance count matches desired capacity' do
+      expected_number = 5
+
+      expect(group).to receive(:desired_capacity).and_return(expected_number)
+      expect(@driver).to receive(:healthy_instance_count).and_return(expected_number)
+
+      expect(@driver.desired_capacity_reached?).to be_true
+    end
+
+    it 'returns false if healthy instance count is less than desired capacity' do
+      expected_number = 5
+
+      expect(group).to receive(:desired_capacity).and_return(expected_number)
+      expect(@driver).to receive(:healthy_instance_count).and_return(expected_number - 1)
+
+      expect(@driver.desired_capacity_reached?).to be_false
+    end
+
+    it 'returns true if healthy instance count is more than desired capacity' do
```

Comment:
The intent of blue/green deployments is to allow for "live" deployments while your application is receiving traffic.  The intent of 'warm_up' therefore, is to insure that your newly deployed stack can handle the same capacity as the existing stack - which is why we set the new stack to have the same number of instances as the existing stack.  Since the assumption is that the same number of instances in the new stack should be able to handle the current load of the old stack, then certainly more instances should be able to handle the current load of the old stack as well.